ogre2/src/Ogre2WideAngleCamera.cc

```diff
@@ -275,22 +339,224 @@ void Ogre2WideAngleCamera::RetrieveCubePassSceneDefs(
+    this->dataPtr->ogreTmpTextures[0], this->dataPtr->ogreTmpTextures[1],
+    this->dataPtr->envCubeMapTexture
+  };
+
+  for (uint32_t i = 0u; i < 6u; ++i)
   {
-    Ogre::CompositorTargetDef *target0 = nodeDef->getTargetPass(i);
-    Ogre::CompositorPassDefVec &passes = target0->getCompositorPassesNonConst();
-    GZ_ASSERT(passes.size() >= 1u,
-              "wide_angle_camera.compositor is out of sync?");
-    GZ_ASSERT(passes[0]->getType() == Ogre::PASS_SCENE,
-              "wide_angle_camera.compositor is out of sync?");
-    GZ_ASSERT(dynamic_cast<Ogre::CompositorPassSceneDef *>(passes[0]),
-              "Memory corruption?");
-
-    this->dataPtr->cubePassSceneDef[i] =
-      static_cast<Ogre::CompositorPassSceneDef *>(passes[0]);
+    GZ_ASSERT(!this->dataPtr->ogreCompositorWorkspace[i], "Must be nullptr!");
+
+    this->dataPtr->ogreCompositorWorkspace[i] = ogreCompMgr->addWorkspace(
+      ogreSceneManager, channels, this->dataPtr->ogreCamera,
+      this->WorkspaceDefinitionName(i), false);
+    this->dataPtr->ogreCompositorWorkspace[i]->addListener(
+      &this->dataPtr->workspaceListener);
+
+    for (RenderPassPtr &pass : this->dataPtr->renderPasses)
+    {
+      Ogre2RenderPass *ogre2RenderPass =
+        dynamic_cast<Ogre2RenderPass *>(pass.get());
+      ogre2RenderPass->WorkspaceAdded(
+        this->dataPtr->ogreCompositorWorkspace[i]);
+    }
   }
 }
 
+//////////////////////////////////////////////////
+void Ogre2WideAngleCamera::DestroyFacesWorkspaces()
+{
+  using namespace Ogre;
+  auto engine = Ogre2RenderEngine::Instance();
+  auto ogreRoot = engine->OgreRoot();
+  Ogre::CompositorManager2 *ogreCompMgr = ogreRoot->getCompositorManager2();
+
+  for (uint32_t i = 0u; i < 6u; ++i)
+  {
+    if (this->dataPtr->ogreCompositorWorkspace[i])
+    {
+      const IdString workspaceName =
+        this->dataPtr->ogreCompositorWorkspace[i]->getDefinition()->getName();
+
+      for (RenderPassPtr &pass : this->dataPtr->renderPasses)
+      {
+        Ogre2RenderPass *ogre2RenderPass =
+          dynamic_cast<Ogre2RenderPass *>(pass.get());
+        ogre2RenderPass->WorkspaceRemoved(
+          this->dataPtr->ogreCompositorWorkspace[i]);
+      }
+      ogreCompMgr->removeWorkspace(this->dataPtr->ogreCompositorWorkspace[i]);
+      this->dataPtr->ogreCompositorWorkspace[i] = nullptr;
+
+      ogreCompMgr->removeWorkspaceDefinition(workspaceName);
+    }
+  }
+  this->dataPtr->cubePassSceneDef = nullptr;
+}
+
+//////////////////////////////////////////////////
+void Ogre2WideAngleCamera::UpdateRenderPasses()
+{
+  using namespace Ogre;
+  bool updateConnection = false;
+
+  // set node instance to render pass and update enabled state
+  for (const auto &pass : this->dataPtr->renderPasses)
+  {
+    Ogre2RenderPass *ogre2RenderPass =
+      dynamic_cast<Ogre2RenderPass *>(pass.get());
+
+    const IdString currNode =
+      ogre2RenderPass->OgreCompositorNodeDefinitionName();
+
+    for (size_t i = 0u; i < 6u; ++i)
```

Comment:
`6u` - > `kWideAngleNumCubemapFaces`?

Reply:
there are also a few other places that can use `kWideAngleNumCubemapFaces` instead of `6u`